dd-sdk-android/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt

```diff
@@ -37,6 +63,40 @@ internal class SessionReplayFeature(coreFeature: CoreFeature) :
         return NoOpDataUploader()
     }
 
+    // endregion
+
+    // region SessionReplayFeature
+
+    internal fun stopRecording() {
+        if (isRecording.getAndSet(false)) {
+            unregisterCallback(appContext)
+        }
+    }
+
+    internal fun startRecording() {
+        if (!isRecording.getAndSet(true)) {
+            registerCallback(appContext)
+        }
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun registerCallback(context: Context) {
+        if (context is Application) {
```

Comment:
we can declare `context` as `Application` type directly and avoid this check

Reply:
Yeah but it doesn't follow the pattern for all the other features. If we do this it will make sense to change this also for `RumFeature`, `LogsFeature`, etc.